cmd/containerd-shim/main_unix.go

```diff
@@ -136,23 +139,37 @@ func serve(server *grpc.Server, path string) error {
 	return nil
 }
 
-func handleSignals(signals chan os.Signal, server *grpc.Server) error {
-	for s := range signals {
-		logrus.WithField("signal", s).Debug("received signal")
-		switch s {
-		case unix.SIGCHLD:
-			if err := reaper.Reap(); err != nil {
-				logrus.WithError(err).Error("reap exit status")
+func handleSignals(signals chan os.Signal, server *grpc.Server, sv *shim.Service) error {
```

Comment:
Seems unnecessary to have an `error` return type. Unless one of the newly added signal propagation calls should return an error?  Or maybe at the very least log the error and remove this return?